inference-engine/ie_bridges/c/tests/ie_c_api_test.cpp

```diff
@@ -281,6 +282,46 @@ TEST(ie_core_read_network, networkRead) {
     ie_core_free(&core);
 }
 
+TEST(ie_core_read_network_from_memory, networkReadFromMemory) {
+    ie_core_t *core = nullptr;
+    IE_ASSERT_OK(ie_core_create("", &core));
+    ASSERT_NE(nullptr, core);
+
+    std::string xml_content;
+    {
+        std::ifstream xml_file(xml);
+        while (xml_file)
+        {
+            char ch;
+            xml_file >> ch;
+            xml_content += ch;
+        }
+    }
+
+    std::vector<uint8_t> model_content;
+    {
+        std::ifstream model_file(bin, std::ifstream::binary);
+        while (model_file)
+        {
+            uint8_t ch((uint8_t)model_file.get());
+            model_content.push_back(ch);
+        }
+    }
+    
+    tensor_desc_t model_desc { C, { 1, { model_content.size() } }, U8 };
+    ie_blob_t *model_blob = nullptr;
+    IE_EXPECT_OK(ie_blob_make_memory_from_preallocated(&model_desc, model_content.data(), model_content.size(), &model_blob));
+    EXPECT_NE(nullptr, model_blob);
+
+    ie_network_t *network = nullptr;
+    IE_EXPECT_OK(ie_core_read_network_from_memory(core, xml_content.data(), model_blob, &network));
+    EXPECT_NE(nullptr, network);
```

Comment:
expect.*** family of checks are not break execution - so the question if you got nullptr here some of further operations doesn't make sense like network free, etc. 
Especially this became a problem in line 312. I Would use asserts of modify fail handling with proper memory releasing. like goto ABORT, etc.

Reply:
I added nullptr check but i'm not sure it's coherent with the rest of the tests in this file which don't do these kinds of checks.